packages/dropdowns/src/Menu/Items/Item.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import StyledItem from '../../styled/items/StyledItem';
+import useDropdownContext from '../../utils/useDropdownContext';
+import useMenuContext from '../../utils/useMenuContext';
+
+// eslint-disable-next-line react/prop-types
+const Item = ({ value, disabled, component = StyledItem, ...props }) => {
+  const {
+    downshift: { selectedItem, highlightedIndex, getItemProps, setHighlightedIndex, itemToString }
+  } = useDropdownContext();
+  const { itemIndexRef } = useMenuContext();
+
+  if (!value && !disabled) {
+    throw new Error('All Item components require a `value` prop');
+  }
+
+  const currentIndex = itemIndexRef.current;
+  const isFocused = highlightedIndex === currentIndex;
+  let isSelected = selectedItem === value;
+
+  if (itemToString) {
+    isSelected = itemToString(selectedItem) === itemToString(value);
+  }
+
+  useEffect(() => {
+    if (!disabled && isSelected && highlightedIndex === null) {
```

Comment:
Should this effect only be run when these things change?

Reply:
Good catch.  I've updated the deps array to include these 3 values.

I've also disabled the `react-hooks/exhaustive-deps` lint rule as these values are the only persistent effects ([Rules of Hooks](https://reactjs.org/docs/hooks-rules.html))